### DISCUSSION TITLE
Need help with showing average data correctly

### DISCUSSION THREAD

**Discussion by JayMcG** [dbuezas/lovelace-plotly-graph-card#310] [Q&A] [answered]
Hey! Loving the Ploty Graph Card! 

I have added a card to show my solar production and grid consumption per day and would like to add average data for the week/month.

My average data is looking skewed as it's including today's information (which is not yet complete). How would I exclude today's data from the calculations? I assume it's as simple as applying a filter.

Thanks in advance! 

![Capture](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/17012684/e378620f-25bc-455d-a126-5bc359f37d7c)

code:

type: custom:plotly-graph
title: 14 Day Energy Analysis
hours_to_show: 14d
defaults:
  yaxes:
    fixedrange: true
entities:
  - entity: sensor.solar_production
    name: Solar
    statistic: state
    period: day
    type: bar
  - entity: sensor.solar_production
    name: Avg. Production
    showlegend: false
    statistic: state
    period: week
    line:
      shape: spline
  - entity: sensor.grid_kwh
    opacity: 0.6
    name: Grid
    statistic: state
    period: day
    type: bar
  - entity: sensor.grid_kwh
    name: Avg. Grid
    showlegend: false
    statistic: state
    period: month
    time_offset: 0d
    line:
      shape: spline
config:
  displayModeBar: false
layout:
  barmode: overlay

**Comment by dbuezas** [comment, marked as answer]
Happy to hear!
Yes, there's a filter called "filter", you can pass it `i<xs.length-1` and that will remove the last datapoint

**Comment by JayMcG** [comment]
Thanks David! 

I added this but it doesn't make any change to the data. I must be doing it incorrectly. 

  - entity: sensor.solar_production
    name: Avg. Prod.
    filters:
      filter: i<xs.length-1
    statistic: state
    period: week

**Reply by dbuezas** [reply]
Please use surround yaml code  like this:
` ```yaml `
` your yaml`
` ``` `

The issue is that the filters are an array, you are missing a minus symbol.
Look in the readme for examples
